test/util/chain.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+const assert = require('bsert');
+const Network = require('../../lib/protocol/network');
+const Chain = require('../../lib/blockchain/chain');
+const Miner = require('../../lib/mining/miner');
+const WorkerPool = require('../../lib/workers/workerpool');
+const rules = require('../../lib/covenants/rules');
+const store = require('../../lib/blockstore');
+const {testdir} = require('./common');
+
+const chainUtils = exports;
```

Comment:
This is cool. We don't have to address in this PR but I can see this module being classified and used throughout the testing suite.